src/main/io/gps_virtual.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+void setVirtualGPS(double latitude, double longitude, double altiutude, double velocity, double velocity3D, double course);
+struct gpsSolutionData_s;
```

Comment:
This way you break dependence on header containing `gpsSolutionData_t`. 

Another option is to `#include` "io/gps.h"` here - the code will be slightly simpler and gps.h would be included in most cases anyway, so it may be a even a bit better. (Sorry, the code looked more complicated with `#ifdefs`s )